modules/features/settings/src/main/java/au/com/shiftyjelly/pocketcasts/settings/history/HistoryFragment.kt

```diff
@@ -59,15 +66,54 @@ class HistoryFragment : BaseFragment(), HasBackstack {
                 composable(HistoryNavRoutes.UpNextHistory) {
                     UpNextHistoryPage(
                         onHistoryEntryClick = { date ->
+                            navController.navigate(
+                                HistoryNavRoutes.upNextHistoryDetailsDestination(
+                                    date.time,
+                                ),
+                            )
                         },
                         onBackClick = navController::popBackStack,
                         bottomInset = bottomInsetDp,
                     )
                 }
+                composable(
+                    HistoryNavRoutes.upNextHistoryDetailsRoute(),
+                    listOf(
+                        navArgument(HistoryNavRoutes.UpNextHistoryDateArgument) {
+                            type = NavType.LongType
+                        },
+                    ),
+                ) { backStackEntry ->
+                    val arguments =
+                        requireNotNull(backStackEntry.arguments) { "Missing back stack entry arguments" }
+                    val date =
+                        requireNotNull(arguments.getLong(HistoryNavRoutes.UpNextHistoryDateArgument)) {
+                            "Missing date argument"
+                        }
+                    UpNextHistoryDetailsPage(
+                        date = date,
+                        onRestoreClick = ::showUpNextRestoreConfirmationDialog,
```

Comment:
This isn't a big deal, so feel free to ignore it, but if I don't have anything currently playing or in my Up Next, the explanation text doesn't really make sense. Maybe it can just be restored. 

<img  width="300" src="https://github.com/user-attachments/assets/4e4b5ef9-f26d-4b61-8960-bb85ced0dca1" />

Reply:
Good point! Why add an extra step? Flow updated in fb33c2e